### PR TITLE
Updating to supply a new endpoint for apis

### PR DESCRIPTION
**Pull request by richardstevens**
This creates a new endpoint for any API (in theory) and is used by the new help and support pages we have.

I have also linted the files based on the new culture guidlines (With the addition of https://github.com/holidayextras/culture/pull/99)

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 machine:
   node:
-    version: 4.4.4
+    version: 6.1.0
 dependencies:
   cache_directories:
     - node_modules
+  override:
+    - npm install -g npm@3.9.0
```
Comment (elliottcrush) [inline circle.yml:8]: Not needed now we're running on `6`

File: src/components/pageData.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        response: newFiles
+      }
+    }
+    for (let i = 0; i < data.length; i++) {
+      let newFile = extraFiles ? { } : clone(this.params.files[ fileName ])
+      let folder = fileName.split('/')
+      folder = (folder && folder.length > 1) ? folder.slice(0, -1).join('/') + '/' : ''
+
+      // Attach page data to props passed to the page
+      let pageData = data[i]
```
Comment (elliottcrush) [inline src/components/pageData.js:34]: Are we extending the current pageData object for ease in the template? Could we get any benefits from attaching a new object?

File: src/components/pageData.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    // This can be prodlib data based on an SEO object
+    if (!fileParams.dataSource.extras) return callBack()
+    const loop = fileParams.dataSource.extras
+    return Object.keys(loop).map(opt => {
+      if (!loop[ opt ].query) return
+      loop[ opt ].extraFiles = true
+      const option = loop[ opt ].query.match(/<%([^%].*)%>/)
+      const extraPageData = data.map(currentFile => {
+        return new Promise((resolve, reject) => {
+          if (!currentFile.data.pageData[option[1]]) return reject(new Error('No extra options found'))
```
Comment (elliottcrush) [inline src/components/pageData.js:106]: Reject should throw the error in the callback, probably just cleaner to pass the message in perhaps

File: circle.yml (hunk 1 above)
Comment (richardstevens) [reply circle.yml:8]: According to circle it uses 2.9 when on v6

File: src/components/pageData.js (hunk 3 above)
Comment (richardstevens) [reply src/components/pageData.js:106]: That doesn't conform to standardjs linting if you dont say its an Error

File: circle.yml (hunk 1 above)
Comment (elliottcrush) [reply circle.yml:8]: Balls

File: src/components/pageData.js (hunk 2 above)
Comment (richardstevens) [reply src/components/pageData.js:34]: I have moved the HX api functionality to share this new apiCaller so am continuing to support pageData, however the extra api calls made will be called whatever the user set in the markdown file.

**Review by elliottcrush — APPROVED** [review]

**Pull request merged**